examples/auction/contract.py

```diff
@@ -17,9 +18,8 @@ def __init__(self) -> None:
         self.auction_end = UInt64(0)
         self.previous_bid = UInt64(0)
         self.asa_amount = UInt64(0)
-        self.asa = Asset(0)
-        # Use zero address rather than an empty string for Account type safety
-        self.previous_bidder = Global.zero_address
+        self.asa = Asset()
+        self.previous_bidder = Account()
```

Comment:
I'm not sold on this change. Could we do `Account.zero_address` ?

Reply:
I think `Account()` as zero address makes sense, but I agree having the more explicit `Account.zero_address` as an option would be more readable as an option?

Reply:
There is a convention in Python where the default/empty constructor yields an empty/"Falsey" value:
- `int() == 0`
- `bool() == False`
- `str() == ""` 
and so on.
As part of this PR, I've made this change to our types, which actually reads a lot nicer where you don't have to explicitly put zero like `Asset(0)` to get an "empty"/`False` `Asset`.
It would be strange if `Account` didn't follow this convention and everything else did.

(side note: I had assumed UInt64 and BigUInt already had these default constructor behaviours for us, just noticed they haven't, adding that now)

Reply:
Another side note @tristanmenzel, with this consistency, we can make a sensible "default empty" constructor for arc4 static arrays too. Will be much easier to do with a bit of EB refactoring too that I'm holding off on until post 1.0